d/example/source/vanillamock/complex.d

```diff
@@ -0,0 +1,22 @@
+// XXX: 3 complex examples, but complexity like this should be desing smells.
+module vanillamock.complex;
+
+// BAD: should re-design to achieve Command-Query separation.
+interface ExampleBothCommandAndQuery
+{
+    int doSomething(bool foo, int bar);
+}
+
+// BAD: should separate into several small interfaces.
+interface ExampleManyMethods
+{
+    void foo(int i);
+    int bar();
+    int baz(int j);
+}
+
+// BAD?: should use promises if available.
```

Comment:
@kubo39 We should use major asynchronous primitives in D like Promises. But I don't know what ones are major in D (probably Task in std.parallelism?).

So, could you modify to use the major async primitives in D at here?